src/controllers/scripting/legacy/scriptconnection.h

```diff
@@ -18,6 +19,7 @@ class ScriptConnection {
     QJSValue callback;
     ControllerScriptInterfaceLegacy* engineJSProxy;
     ControllerScriptEngineLegacy* controllerEngine;
+    CompressingProxy* proxy;
```

Comment:
This plain pointer should be replaced with std::unique_ptr

Reply:
This suggestion looked easy and obvious.
But in practice I struggle with changeing this pointer to a smartpointer. This seems to be, because ScriptConnection connection is passed at many unrelated places of the code by reference. Maybe there is a simple solution for this, but for me it looks like, that it requires to change all occurences of passing by reference to passing by pointer.

Do you think it's worth to change much unrelated code for this?

Reply:
If this plain pointer gets copied/cloned in uncontrolled ways then we have a serious issue.

Reply:
Can you elaborate?
Why do you need to change how the `ScriptConnection` is passed because a member changed?

Reply:
Passing a reference is ok, copying is not.

Reply:
The lifetime of the proxy is undefined. It is managed by the add/removeScriptConnection() calls that are not guaranteed to occur in pairs.

Reply:
@Swiftb0y
> Can you elaborate?
> Why do you need to change how the `ScriptConnection` is passed because a member changed?

To be honest, I don't understood what happens. But here is what happens: I added an EXAMPLE smartpointer to ScriptConnection class (only this one line):
![grafik](https://user-images.githubusercontent.com/64457745/145729957-21d3bca4-6a46-48a8-bb70-ecd6bfb5d017.png)

And when I compile the code I get:
```
  [3/13] Automatic MOC and UIC for target mixxx-lib
  [4/13] Building CXX object CMakeFiles\mixxx-lib.dir\src\control\controlobjectscript.cpp.obj
  FAILED: CMakeFiles/mixxx-lib.dir/src/control/controlobjectscript.cpp.obj 
  C:\PROGRA~2\MICROS~3\2019\COMMUN~1\VC\Tools\MSVC\1429~1.301\bin\Hostx64\x64\cl.exe  /nologo /TP -DAMD64 -DHAVE_WINSOCK2_H -DMIXXX_BUILD_RELEASE -DMIXXX_DEBUG_ASSERTIONS_ENABLED -DMIXXX_DEBUG_ASSERTIONS_FATAL -DNDEBUG -DNOMINMAX -DQT_CONCURRENT_LIB -DQT_CORE_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_NO_DEBUG -DQT_OPENGL_LIB -DQT_PRINTSUPPORT_LIB -DQT_QMLMODELS_LIB -DQT_QML_LIB -DQT_QUICKWIDGETS_LIB -DQT_QUICK_LIB -DQT_SQL_LIB -DQT_SVG_LIB -DQT_TABLET_SUPPORT -DQT_TESTCASE_BUILDDIR=\"C:/Users/Joerg/source/repos/JoergAtGithub/mixxx/build/x64__portable\" -DQT_TESTLIB_LIB -DQT_USE_QSTRINGBUILDER -DQT_WIDGETS_LIB -DQT_XML_LIB -DSFC_SUPPORTS_SET_COMPRESSION_LEVEL -DUNICODE -DWIN32 -DWINVER=0x0601 -D_ATL_MIN_CRT -D_CRT_SECURE_NO_WARNINGS -D_SILENCE_CXX17_ITERATOR_BASE_CLASS_DEPRECATION_WARNING -D_USE_MATH_DEFINES -D_WIN32_WINNT=0x0601 -D__BATTERY__ -D__BROADCAST__ -D__ENGINEPRIME__ -D__HID__ -D__HSS1394__ -D__LILV__ -D__MAD__ -D__MEDIAFOUNDATION__ -D__MODPLUG__ -D__OPUS__ -D__QTKEYCHAIN__ -D__SNDFILE__ -D__SQLITE3__ -D__SSE2__ -D__SSE__ -D__VINYLCONTROL__ -D__WINDOWS__ -D__WV__ -Imixxx-lib_autogen\include -I..\..\src -Isrc -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\include -I..\..\lib\fidlib -I..\..\lib\googletest\googletest\include -I..\..\lib\portaudio -I..\..\lib\rigtorp\SPSCQueue\include -I..\..\lib\replaygain -I..\..\lib\reverb -I..\..\lib\libshout-idjc\include -I..\..\lib\kaitai -I..\..\lib\mp3guessenc-0.27.4 -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\include\qt5 -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\include\qt5\QtConcurrent -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\include\qt5\QtCore -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\tools\qt5\mkspecs\win32-msvc -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\include\qt5\QtGui -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\include\qt5\QtNetwork -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\include\qt5\QtOpenGL -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\include\qt5\QtWidgets -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\include\qt5\QtPrintSupport -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\include\qt5\QtQml -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\include\qt5\QtQuickWidgets -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\include\qt5\QtQuick -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\include\qt5\QtQmlModels -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\include\qt5\QtSql -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\include\qt5\QtSvg -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\include\qt5\QtTest -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\include\qt5\QtXml -I..\..\lib\qm-dsp -I..\..\lib\qm-dsp\include -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\include\taglib -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\include\opus -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\include\hidapi -I..\..\lib\xwax /DWIN32 /D_WINDOWS /GR /EHsc /Zc:__cplusplus /Z7 /O2 /Ob1 /DNDEBUG -MD /UTF8 /fp:fast /Gy /W3 -std:c++17 /showIncludes /FoCMakeFiles\mixxx-lib.dir\src\control\controlobjectscript.cpp.obj /FdCMakeFiles\mixxx-lib.dir\mixxx-lib.pdb /FS -c ..\..\src\control\controlobjectscript.cpp
C:\Users\Joerg\source\repos\JoergAtGithub\mixxx\src\control\controlobjectscript.h(25): error C2280: 'ScriptConnection::ScriptConnection(const ScriptConnection &)': attempting to reference a deleted function
  C:\Users\Joerg\source\repos\JoergAtGithub\mixxx\src\controllers/scripting/legacy/scriptconnection.h(34): note: compiler has generated 'ScriptConnection::ScriptConnection' here
  C:\Users\Joerg\source\repos\JoergAtGithub\mixxx\src\controllers/scripting/legacy/scriptconnection.h(34): note: 'ScriptConnection::ScriptConnection(const ScriptConnection &)': function was implicitly deleted because a data member invokes a deleted or inaccessible function 'std::unique_ptr<int,std::default_delete<int>>::unique_ptr(const std::unique_ptr<int,std::default_delete<int>> &)'
  C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\memory(3269): note: 'std::unique_ptr<int,std::default_delete<int>>::unique_ptr(const std::unique_ptr<int,std::default_delete<int>> &)': function was explicitly deleted
  [5/13] Building CXX object CMakeFiles\mixxx-lib.dir\src\controllers\scripting\legacy\scriptconnectionjsproxy.cpp.obj
  FAILED: CMakeFiles/mixxx-lib.dir/src/controllers/scripting/legacy/scriptconnectionjsproxy.cpp.obj 
  C:\PROGRA~2\MICROS~3\2019\COMMUN~1\VC\Tools\MSVC\1429~1.301\bin\Hostx64\x64\cl.exe  /nologo /TP -DAMD64 -DHAVE_WINSOCK2_H -DMIXXX_BUILD_RELEASE -DMIXXX_DEBUG_ASSERTIONS_ENABLED -DMIXXX_DEBUG_ASSERTIONS_FATAL -DNDEBUG -DNOMINMAX -DQT_CONCURRENT_LIB -DQT_CORE_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_NO_DEBUG -DQT_OPENGL_LIB -DQT_PRINTSUPPORT_LIB -DQT_QMLMODELS_LIB -DQT_QML_LIB -DQT_QUICKWIDGETS_LIB -DQT_QUICK_LIB -DQT_SQL_LIB -DQT_SVG_LIB -DQT_TABLET_SUPPORT -DQT_TESTCASE_BUILDDIR=\"C:/Users/Joerg/source/repos/JoergAtGithub/mixxx/build/x64__portable\" -DQT_TESTLIB_LIB -DQT_USE_QSTRINGBUILDER -DQT_WIDGETS_LIB -DQT_XML_LIB -DSFC_SUPPORTS_SET_COMPRESSION_LEVEL -DUNICODE -DWIN32 -DWINVER=0x0601 -D_ATL_MIN_CRT -D_CRT_SECURE_NO_WARNINGS -D_SILENCE_CXX17_ITERATOR_BASE_CLASS_DEPRECATION_WARNING -D_USE_MATH_DEFINES -D_WIN32_WINNT=0x0601 -D__BATTERY__ -D__BROADCAST__ -D__ENGINEPRIME__ -D__HID__ -D__HSS1394__ -D__LILV__ -D__MAD__ -D__MEDIAFOUNDATION__ -D__MODPLUG__ -D__OPUS__ -D__QTKEYCHAIN__ -D__SNDFILE__ -D__SQLITE3__ -D__SSE2__ -D__SSE__ -D__VINYLCONTROL__ -D__WINDOWS__ -D__WV__ -Imixxx-lib_autogen\include -I..\..\src -Isrc -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\include -I..\..\lib\fidlib -I..\..\lib\googletest\googletest\include -I..\..\lib\portaudio -I..\..\lib\rigtorp\SPSCQueue\include -I..\..\lib\replaygain -I..\..\lib\reverb -I..\..\lib\libshout-idjc\include -I..\..\lib\kaitai -I..\..\lib\mp3guessenc-0.27.4 -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\include\qt5 -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\include\qt5\QtConcurrent -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\include\qt5\QtCore -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\tools\qt5\mkspecs\win32-msvc -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\include\qt5\QtGui -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\include\qt5\QtNetwork -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\include\qt5\QtOpenGL -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\include\qt5\QtWidgets -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\include\qt5\QtPrintSupport -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\include\qt5\QtQml -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\include\qt5\QtQuickWidgets -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\include\qt5\QtQuick -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\include\qt5\QtQmlModels -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\include\qt5\QtSql -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\include\qt5\QtSvg -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\include\qt5\QtTest -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\include\qt5\QtXml -I..\..\lib\qm-dsp -I..\..\lib\qm-dsp\include -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\include\taglib -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\include\opus -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\include\hidapi -I..\..\lib\xwax /DWIN32 /D_WINDOWS /GR /EHsc /Zc:__cplusplus /Z7 /O2 /Ob1 /DNDEBUG -MD /UTF8 /fp:fast /Gy /W3 -std:c++17 /showIncludes /FoCMakeFiles\mixxx-lib.dir\src\controllers\scripting\legacy\scriptconnectionjsproxy.cpp.obj /FdCMakeFiles\mixxx-lib.dir\mixxx-lib.pdb /FS -c ..\..\src\controllers\scripting\legacy\scriptconnectionjsproxy.cpp
C:\Users\Joerg\source\repos\JoergAtGithub\mixxx\src\controllers\scripting\legacy\scriptconnectionjsproxy.h(14): error C2280: 'ScriptConnection &ScriptConnection::operator =(const ScriptConnection &)': attempting to reference a deleted function
  C:\Users\Joerg\source\repos\JoergAtGithub\mixxx\src\controllers/scripting/legacy/scriptconnection.h(34): note: compiler has generated 'ScriptConnection::operator =' here
  C:\Users\Joerg\source\repos\JoergAtGithub\mixxx\src\controllers/scripting/legacy/scriptconnection.h(34): note: 'ScriptConnection &ScriptConnection::operator =(const ScriptConnection &)': function was implicitly deleted because a data member invokes a deleted or inaccessible function 'std::unique_ptr<int,std::default_delete<int>> &std::unique_ptr<int,std::default_delete<int>>::operator =(const std::unique_ptr<int,std::default_delete<int>> &)'
  C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\memory(3270): note: 'std::unique_ptr<int,std::default_delete<int>> &std::unique_ptr<int,std::default_delete<int>>::operator =(const std::unique_ptr<int,std::default_delete<int>> &)': function was explicitly deleted
  [6/13] Building CXX object CMakeFiles\mixxx-lib.dir\src\controllers\scripting\legacy\scriptconnection.cpp.obj
  [7/13] Building CXX object CMakeFiles\mixxx-lib.dir\src\controllers\scripting\legacy\controllerscriptinterfacelegacy.cpp.obj
  FAILED: CMakeFiles/mixxx-lib.dir/src/controllers/scripting/legacy/controllerscriptinterfacelegacy.cpp.obj 
  C:\PROGRA~2\MICROS~3\2019\COMMUN~1\VC\Tools\MSVC\1429~1.301\bin\Hostx64\x64\cl.exe  /nologo /TP -DAMD64 -DHAVE_WINSOCK2_H -DMIXXX_BUILD_RELEASE -DMIXXX_DEBUG_ASSERTIONS_ENABLED -DMIXXX_DEBUG_ASSERTIONS_FATAL -DNDEBUG -DNOMINMAX -DQT_CONCURRENT_LIB -DQT_CORE_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_NO_DEBUG -DQT_OPENGL_LIB -DQT_PRINTSUPPORT_LIB -DQT_QMLMODELS_LIB -DQT_QML_LIB -DQT_QUICKWIDGETS_LIB -DQT_QUICK_LIB -DQT_SQL_LIB -DQT_SVG_LIB -DQT_TABLET_SUPPORT -DQT_TESTCASE_BUILDDIR=\"C:/Users/Joerg/source/repos/JoergAtGithub/mixxx/build/x64__portable\" -DQT_TESTLIB_LIB -DQT_USE_QSTRINGBUILDER -DQT_WIDGETS_LIB -DQT_XML_LIB -DSFC_SUPPORTS_SET_COMPRESSION_LEVEL -DUNICODE -DWIN32 -DWINVER=0x0601 -D_ATL_MIN_CRT -D_CRT_SECURE_NO_WARNINGS -D_SILENCE_CXX17_ITERATOR_BASE_CLASS_DEPRECATION_WARNING -D_USE_MATH_DEFINES -D_WIN32_WINNT=0x0601 -D__BATTERY__ -D__BROADCAST__ -D__ENGINEPRIME__ -D__HID__ -D__HSS1394__ -D__LILV__ -D__MAD__ -D__MEDIAFOUNDATION__ -D__MODPLUG__ -D__OPUS__ -D__QTKEYCHAIN__ -D__SNDFILE__ -D__SQLITE3__ -D__SSE2__ -D__SSE__ -D__VINYLCONTROL__ -D__WINDOWS__ -D__WV__ -Imixxx-lib_autogen\include -I..\..\src -Isrc -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\include -I..\..\lib\fidlib -I..\..\lib\googletest\googletest\include -I..\..\lib\portaudio -I..\..\lib\rigtorp\SPSCQueue\include -I..\..\lib\replaygain -I..\..\lib\reverb -I..\..\lib\libshout-idjc\include -I..\..\lib\kaitai -I..\..\lib\mp3guessenc-0.27.4 -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\include\qt5 -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\include\qt5\QtConcurrent -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\include\qt5\QtCore -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\tools\qt5\mkspecs\win32-msvc -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\include\qt5\QtGui -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\include\qt5\QtNetwork -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\include\qt5\QtOpenGL -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\include\qt5\QtWidgets -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\include\qt5\QtPrintSupport -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\include\qt5\QtQml -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\include\qt5\QtQuickWidgets -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\include\qt5\QtQuick -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\include\qt5\QtQmlModels -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\include\qt5\QtSql -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\include\qt5\QtSvg -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\include\qt5\QtTest -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\include\qt5\QtXml -I..\..\lib\qm-dsp -I..\..\lib\qm-dsp\include -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\include\taglib -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\include\opus -I..\..\buildenv\mixxx-deps-2.4-x64-windows-058063e\installed\x64-windows\include\hidapi -I..\..\lib\xwax /DWIN32 /D_WINDOWS /GR /EHsc /Zc:__cplusplus /Z7 /O2 /Ob1 /DNDEBUG -MD /UTF8 /fp:fast /Gy /W3 -std:c++17 /showIncludes /FoCMakeFiles\mixxx-lib.dir\src\controllers\scripting\legacy\controllerscriptinterfacelegacy.cpp.obj /FdCMakeFiles\mixxx-lib.dir\mixxx-lib.pdb /FS -c ..\..\src\controllers\scripting\legacy\controllerscriptinterfacelegacy.cpp
C:\Users\Joerg\source\repos\JoergAtGithub\mixxx\src\control\controlobjectscript.h(25): error C2280: 'ScriptConnection::ScriptConnection(const ScriptConnection &)': attempting to reference a deleted function
  C:\Users\Joerg\source\repos\JoergAtGithub\mixxx\src\controllers/scripting/legacy/scriptconnection.h(34): note: compiler has generated 'ScriptConnection::ScriptConnection' here
  C:\Users\Joerg\source\repos\JoergAtGithub\mixxx\src\controllers/scripting/legacy/scriptconnection.h(34): note: 'ScriptConnection::ScriptConnection(const ScriptConnection &)': function was implicitly deleted because a data member invokes a deleted or inaccessible function 'std::unique_ptr<int,std::default_delete<int>>::unique_ptr(const std::unique_ptr<int,std::default_delete<int>> &)'
  C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\memory(3269): note: 'std::unique_ptr<int,std::default_delete<int>>::unique_ptr(const std::unique_ptr<int,std::default_delete<int>> &)': function was explicitly deleted
C:\Users\Joerg\source\repos\JoergAtGithub\mixxx\src\controllers\scripting\legacy\scriptconnectionjsproxy.h(14): error C2280: 'ScriptConnection &ScriptConnection::operator =(const ScriptConnection &)': attempting to reference a deleted function
  C:\Users\Joerg\source\repos\JoergAtGithub\mixxx\src\controllers/scripting/legacy/scriptconnection.h(34): note: compiler has generated 'ScriptConnection::operator =' here
  C:\Users\Joerg\source\repos\JoergAtGithub\mixxx\src\controllers/scripting/legacy/scriptconnection.h(34): note: 'ScriptConnection &ScriptConnection::operator =(const ScriptConnection &)': function was implicitly deleted because a data member invokes a deleted or inaccessible function 'std::unique_ptr<int,std::default_delete<int>> &std::unique_ptr<int,std::default_delete<int>>::operator =(const std::unique_ptr<int,std::default_delete<int>> &)'
  C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\memory(3270): note: 'std::unique_ptr<int,std::default_delete<int>> &std::unique_ptr<int,std::default_delete<int>>::operator =(const std::unique_ptr<int,std::default_delete<int>> &)': function was explicitly deleted
  [8/13] Building CXX object CMakeFiles\mixxx-lib.dir\mixxx-lib_autogen\mocs_compilation.cpp.obj
  ninja: build stopped: subcommand failed.
```

Reply:
Thank the compiler to tell you that your code is unsound. Replacing the plain pointer with a managed pointer just revealed those issues. Yes, it might still work at runtime but I prefer to write code that is safe and sound at compile time!

Switch to Rust where code is safe and sound by default.

Reply:
It looks like that the original compiler error is not part of the log snipped above. My guess is that it fails, because you cannot longer copy an object that contains an unique pointer.

To solve this issue you need to make sure the ScriptConnection a s not copied. 

I have not yet looked into the details, but 
as a start u suggest to delete to delete the auto generated copy constructor. 

Due to the connections, the object should have probably never been copied so this PR only discovers an existing issue.

Reply:
If preventing copies doesn't work the plain pointer needs to initialized and reset to `nullptr` appropriately with lots of assertions and this behavior needs to be documented in every detail.

Reply:
I found a much better fittig place to store this pointer, which is never copied. I could changed the code to a std:unique_ptr therefore.